src/Expr.cc

```diff
@@ -4319,6 +4320,52 @@ void CallExpr::ExprDescribe(ODesc* d) const
 		args->Describe(d);
 	}
 
+LambdaExpr::LambdaExpr(std::unique_ptr<function_ingredients> ingredients,
+		       std::shared_ptr<id_list> outer_ids)
+	{
+	this->ingredients = std::move(ingredients);
+	this->outer_ids = std::move(outer_ids);
+	SetType(this->ingredients->id->Type()->Ref());
+	}
+
+Val* LambdaExpr::Eval(Frame* f) const
+	{
+	BroFunc* lamb = new BroFunc(
+		ingredients->id,
+		ingredients->body,
+		ingredients->inits,
+		ingredients->frame_size,
+		ingredients->priority);
+
+	lamb->AddClosure(outer_ids, f);
+
+	ingredients->id->SetVal((new Val(lamb))->Ref());
+	ingredients->id->SetConst();
+	ingredients->id->ID_Val()->AsFunc()->SetScope(ingredients->scope);
+
+	return ingredients->id->ID_Val();
+	}
+
+void LambdaExpr::ExprDescribe(ODesc* d) const
```

Comment:
This should render back Bro script code, at least when in "readable" mode. Take a look at some of the other `ExprDescribe` methods.